### PR TITLE
#66 test and fix

### PR DESCRIPTION
**Pull request by Ekstazi**

### CODE REVIEW

File: lib/Worker/DefaultPool.php [hunk 1]
```diff
@@ -68,8 +68,6 @@ public function __construct(int $maxSize = self::DEFAULT_MAX_SIZE, WorkerFactory
         $busyQueue = $this->busyQueue;
 
         $this->push = static function (Worker $worker) use ($workers, $idleWorkers, $busyQueue) {
-            \assert($workers->contains($worker), "The provided worker was not part of this queue");
-
```
Comment (kelunik) [inline lib/Worker/DefaultPool.php]: Why did you remove this assertion?

Comment (Ekstazi) [reply lib/Worker/DefaultPool.php]: See issue #66 for more details.
1. Pool autoremove all stopped workers.
2. PooledWorker::_destroy return worker back to pool
3. Because of that assert will fail and phpunit tests also.

Comment (kelunik) [reply lib/Worker/DefaultPool.php]: Maybe we shouldn't auto-remove all stopped workers?

**Comment by Ekstazi** [comment]
No. It's good feature. I use it.

пн, 17 дек. 2018 г., 22:33 Niklas Keller notifications@github.com:

**Comment by trowski** [comment]
Manually merged test and pushed a slightly different fix in 1c9822a564219fd9ab0de43bb80455d41154d519.

**Comment by Ekstazi** [comment]
thx
